libs/domain/product/src/mixins/product.mixin.ts

```diff
@@ -34,13 +36,19 @@ export const ProductMixin = <
 
     protected contextController = new ContextController(this);
 
-    protected $productContext = signal(
-      this.contextController.get<string>(ProductContext.SKU)
+    protected $productQualifier: Signal<ProductQualifier | undefined> = signal(
+      this.contextController
+        .get<ProductQualifier>(ProductContext.SKU)
+        .pipe(
+          map((sku) =>
+            featureVersion >= '1.3' ? sku : ({ sku } as ProductQualifier)
+          )
+        )
     );
 
     protected $product = computed(() => {
-      const sku = this.sku ?? this.$productContext();
-      return sku ? this.productService?.get({ sku }) : of(null);
+      const qualifier = this.sku ? { sku: this.sku } : this.$productQualifier();
```

Comment:
we can discuss, if we should `protected $productContext`, but because it was not exposed in an mixin interface, I decided to treat it as a private API